src/utils/ChaiUtils.ts

```diff
@@ -34,13 +34,15 @@ export namespace ChaiUtils {
         while (leftSide !== undefined) {
             if (leftSide.kind === ts.SyntaxKind.CallExpression) {
                 return <ts.CallExpression>leftSide;
-            } else if (leftSide.kind === ts.SyntaxKind.PropertyAccessExpression) {
+            }
+
+            if (leftSide.kind === ts.SyntaxKind.PropertyAccessExpression) {
                 leftSide = (<ts.PropertyAccessExpression>leftSide).expression;
             } else if (checkParent && leftSide.parent.kind === ts.SyntaxKind.CallExpression) {
                 return <ts.CallExpression>leftSide.parent;
-            } else {
-                return undefined; // cannot walk any further left in the property expression
             }
+
+            return undefined; // cannot walk any further left in the property expression
```

Comment:
This one looks suspicious.
Because logic above has case `if (leftSide.kind === ts.SyntaxKind.PropertyAccessExpression) {` that doesn't return anything, but if I understand correctly - this `return` will always triggered after first iteration where `leftSide.kind === ts.SyntaxKind.PropertyAccessExpression`

Looks like  `if (leftSide.kind === ts.SyntaxKind.PropertyAccessExpression) {` at least should have `continue;` statement to work correctly or code should be reordered to perform checks with all `return`'s above `leftSide` reassignment.

Reply:
Ha, good catch!